### ISSUE TITLE
Error when setting `executor` in `@ct.lattice`: `DaskExecutor` object has no attribute `scheduler_address` 

### ISSUE THREAD

**Issue by Andrew-S-Rosen** [AgnostiqHQ/covalent#1781]
### Environment

- **Covalent version**: >=0.227.0rc0
- **Python version**: 3.10.13
- **Operating system**: Linux

### What is happening?

When setting the `executor` kwarg in the `@ct.lattice` decorator, I receive the following error message in the post-processing step of the workflow.

```bas
'DaskExecutor' object has no attribute 'scheduler_address'
```

**Note**: setting the `executor` kwarg in the `@ct.electron` (but not `@ct.lattice`) works fine and the workflow goes to completion. This seems very closely related to #1733, which is perhaps not fully solved.

![image](https://github.com/AgnostiqHQ/covalent/assets/8674072/609fb67e-be72-4749-b34d-835ec45bf05d)

### How can we reproduce the issue?

```python
import covalent as ct

executor = ct.executor.LocalExecutor()

@ct.electron
def add(a, b):
    return a + b

@ct.lattice(executor=executor)
def workflow(a, b):
    return add(a, b)


dispatch_id = ct.dispatch(workflow)(1, 2)
result = ct.get_result(dispatch_id,wait=True)
assert result.status == "COMPLETED"
```

### What should happen?

The workflow post-processing step should complete.

### Any suggestions?

I need to see if the same behavior occurs on `main`.

**Comment by santoshkumarradha** [comment]
Thank you for the detailed bug report @Andrew-S-Rosen  and for taking the time to include a comprehensive example! Your issue is well-explained, and the provided code snippet is immensely helpful for debugging.

It appears that setting the executor keyword argument in the @ct.lattice decorator is causing unintended behavior in the post-processing step.  It seems like setting the executor attribute within the decorator may inadvertently be modifying the value of the workflow_executor parameter, and thus causing the error you're observing. This does look like a bug on our end.

a temporary workaround would be to explicitly set the workflow_executor parameter when the lattice level executor is overridden. Thank you for providing that solution as well. example - 

```python
import covalent as ct

executor = ct.executor.LocalExecutor()

@ct.electron
def add(a, b):
    return a + b

@ct.lattice(executor=executor, workflow_executor=executor)
def workflow(a, b):
    return add(a, b)

dispatch_id = ct.dispatch(workflow)(1, 2)
result = ct.get_result(dispatch_id, wait=True)
```

But this is indeed a bug, we will tag it along for a solution.

**Comment by Andrew-S-Rosen** [comment]
Thank you for the insight from your perspective! That does seem very reasonable, and I appreciate the potential workaround. I will give that a go and see how things fare! Much appreciated, as always!

**Comment by Andrew-S-Rosen** [comment]
I can confirm that the error was introduced between `0.226.0rc0` and `0.227.0rc0`, as the former works while the latter does not (same issue in `0.228.0rc0`). This should help in narrowing it down! I also provided a more user-friendly minimal example.

**Comment by Andrew-S-Rosen** [comment]
@wjcunningham7: I've narrowed down the location of the issue. It was from commit ad5b742c942a467a8ecd96a993cc9684ab9be307. Specifically, it has to do with the `executor_data` field. If you remove it, the workflow completes successfully. If you add it back in (as in that commit), the post-process step fails.

**Comment by Andrew-S-Rosen** [comment]
It seems like the following lines may be part of the problem, but I still haven't quite sorted it out:

https://github.com/AgnostiqHQ/covalent/blob/bfe3c3528814bc7b1c13f2fed809bf6cc71c0c41/covalent/_workflow/electron.py#L378-L386

**Comment by cjao** [comment]
>Edit: I feel like it might have to do with this [block](https://github.com/AgnostiqHQ/covalent/blob/bfe3c3528814bc7b1c13f2fed809bf6cc71c0c41/covalent/_workflow/electron.py#L378-L386).

@Andrew-S-Rosen You're absolutely right. Will be fixed in #1849 .
